src/plugins.c

```diff
@@ -562,6 +705,8 @@ plugin_free(Plugin *plugin)
 
 	plugin_list = g_list_remove(plugin_list, plugin);
 
+	if (plugin->cb_data_destroy)
+		plugin->cb_data_destroy(plugin->cb_data);
```

Comment:
I managed to lock-up and segfault Geany because of this. I'm not sure exactly why, but I suspect it's because `plugin_free()` is called from `pm_plugin_toggled()` before a plugin is fully setup. It might need to check if the plugin is currently loaded before accessing `cb_data_destroy` and/or `cb_data` members.

In my code I was calling `geany_plugin_set_data()` directly after `GEANY_PLUGIN_REGISTER()` in the `geany_load_module()` hook.